src/index.ts

```diff
@@ -101,9 +102,24 @@ const getNamedType = (
                             foundType.types && foundType.types[0],
                         );
                     case 'scalar':
-                        // it's a scalar, let's use a string as a value.
-                        // This could be improved with a custom scalar definition in the config
-                        return `'${casual.word}'`;
+                        // it's a scalar, let's use a string as a value if there is no custom
+                        // mapping for this particular scalar
+                        if (!customScalars || !customScalars[foundType.name]) {
+                            return `'${casual.word}'`;
+                        }
+
+                        // If there is a mapping to a `casual` type, then use it and make sure
+                        // to call it if it's a function
+                        const embeddedGenerator = casual[customScalars[foundType.name]];
+                        const value = typeof embeddedGenerator === 'function' ? embeddedGenerator() : embeddedGenerator;
+
+                        if (typeof value === 'string') {
+                            return `'${value}'`;
+                        }
+                        if (typeof value === 'object') {
+                            return `${JSON.stringify(value)}`;
```

Comment:
@ardeois added `JSON.stringify` to handle autogenerated types of array/object.

Those will not normally appear since this handler has to do with a particular Scalar (and never an array of a Scalar), but still nothing stops the users from specifying `casual.word_list` which would cause issues.

All in all, this handles an edge case where user input is not following the schema. If you feel it's unneeded, I can remove it